source/crd_test.go

```diff
@@ -54,20 +55,21 @@ func objBody(codec runtime.Codec, obj runtime.Object) io.ReadCloser {
 	return ioutil.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(codec, obj))))
 }
 
-func startCRDServerToServeTargets(endpoints []*endpoint.Endpoint, apiVersion, kind, namespace, name string) rest.Interface {
+func startCRDServerToServeTargets(endpoints []*endpoint.Endpoint, apiVersion, kind, namespace, name string, t *testing.T) rest.Interface {
 	groupVersion, _ := schema.ParseGroupVersion(apiVersion)
 	scheme := runtime.NewScheme()
 	addKnownTypes(scheme, groupVersion)
 
 	dnsEndpointList := endpoint.DNSEndpointList{}
-	dnsEndpoint := endpoint.DNSEndpoint{
+	dnsEndpoint := &endpoint.DNSEndpoint{
 		TypeMeta: metav1.TypeMeta{
```

Comment:
The pointer is not needed here

Reply:
Could you share any more details here?
I think we need it because the PUT request will change the content of dnsEndpoint and return to GET request later for test case.